### PR TITLE
Standards updates 2019 part 2

### PR DESCRIPTION
**Pull request by losborne**
Pull requests into cql-execution require the following.
Submitter and reviewer should ✔ when done.
For items that are not-applicable, mark "N/A" and ✔.
More updates for CQL 1.4

The Expand PR didn't make it into this (yet?)

[CDS Connect](https://cds.ahrq.gov/cdsconnect) and [Bonnie](https://github.com/projecttacoma/bonnie) are the main users of this repository. 
It is strongly recommended to include a person from each of those projects as a reviewer.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Code diff has been done and been reviewed (it does not contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] Code coverage has not gone down and all code touched or added is covered.
- [x] All dependent libraries are appropriately updated or have a corresponding PR related to this change
- [x] `cql4browsers.js` built with `yarn run build-everything` if coffeescript source changed.

**Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: src/elm/arithmetic.coffee [hunk 1]
```diff
@@ -243,18 +274,38 @@ module.exports.Successor = class Successor extends Expression
...
   exec: (ctx) ->
     arg = @execArgs(ctx)
+    successor = null
     if (not arg?)
       null
     else
-      MathUtil.successor arg
+      try
+        # MathUtil.successor throws on underflow, and the exception is used in
+        # the logic for evaluating `meets`
```
Comment (cmoesel) [inline src/elm/arithmetic.coffee]: If this is the case, shouldn't you be re-throwing the exception below?

File: src/elm/arithmetic.coffee [hunk 2]
```diff
@@ -243,18 +274,38 @@ module.exports.Successor = class Successor extends Expression
...
+        successor = MathUtil.successor arg
+      catch e
+        if e instanceof MathUtil.OverFlowException
+          null
```
Comment (cmoesel) [inline src/elm/arithmetic.coffee]: Again, based on the comment above, I don't know if we should actually be re-throwing this, but... as the code is now, I don't see the purpose of the `null` line above.  It's not the last line in the function, so it will not return.  And it is not assigned to anything.  So... it seems like a no-op?

File: src/elm/arithmetic.coffee [hunk 3]
```diff
@@ -243,18 +274,38 @@ module.exports.Successor = class Successor extends Expression
...
       null
     else
-      MathUtil.predecessor arg
+      # MathUtil.predecessor throws on underflow, and the exception is used in
+      # the logic for evaluating `meets`
+      try
+        predecessor = MathUtil.predecessor arg
+      catch e
+        if e instanceof MathUtil.OverFlowException
+          null
```
Comment (cmoesel) [inline src/elm/arithmetic.coffee]: Same comments as above apply here too.

File: src/elm/arithmetic.coffee (hunk 1 above)
Comment (losborne) [reply src/elm/arithmetic.coffee]: The comment was meant to indicate why we are `try..catch`ing rather than just adjusting `MathUtil.successor` to return `null` instead of throwing.  I'll update the comment to hopefully be clearer

File: src/elm/arithmetic.coffee (hunk 2 above)
Comment (losborne) [reply src/elm/arithmetic.coffee]: Good catch--it should be `return null` here.

File: src/elm/arithmetic.coffee (hunk 1 above)
Comment (cmoesel) [reply src/elm/arithmetic.coffee]: Oh right.  I get it now.  The other code counts on `MathUtil.successor` throwing, _not_ *this* code throwing...  Sorry for the mixup!

**Comment by cmoesel** [comment]
I just rebased to cleanup the commits and remove the duplicate commits that must have somehow got in there through a merge.  I did a byte-by-byte diff between the original branch and my rebased branch and confirmed the contents are exactly the same.
